### ISSUE TITLE
ADIOS Particles are not fully openPMD-compatible

### ISSUE THREAD

**Issue by KseniaBastrakova** [ComputationalRadiationPhysics/picongpu#3119]
I create ADIOS checkpoints during the simulation.

` TBG_checkpoint="--checkpoint.period 1000 --checkpoint.backend adios"`
I got some ADIOS file, as a result.
```
checkpoint_0.bp                 
checkpoint_0.bp.dir             
checkpoint_1000.bp              
checkpoint_1000.bp.dir          
checkpoint_2000.bp
.....
```
When I tryed open one of ADIOS file, i got error:
`series = openpmd_api.Series(checkpoint_0.bp, openpmd_api.Access_Type.read_only)`

> 
> Series constructor called with explicit iteration suggests loading a single file with groupBased iteration encoding. Loaded file is fileBased.
> Traceback (most recent call last):
>   File "hdf_to_gdf.py", line 488, in <module>
>     hdf_to_gdf(args.hdf, args.gdf, args.max_cell, args.species)
>   File "hdf_to_gdf.py", line 31, in hdf_to_gdf
>     series_hdf = openpmd_api.Series(hdf_file_directory, openpmd_api.Access_Type.read_only)
> RuntimeError: Unexpected Attribute datatype for 'timeOffset'

i'm using ADIOS 1.13.1
And openPMD latest (d68e9fc2),
PicOnGPU latest (d340812)

**Comment by sbastrakov** [comment]
@KseniaBastrakova as far as I understand, with HDF5 checkpoints it all works fine?

**Comment by ax3l** [comment]
Oh dang, I think the `7` should read `1` here:
https://github.com/ComputationalRadiationPhysics/picongpu/blob/5373b6e64decb304f42e25a4b69f06e51d38222a/include/picongpu/plugins/adios/writer/ParticleAttributeSize.hpp#L139

**Comment by PrometheusPi** [comment]
Great finding @KseniaBastrakova :+1: .
I assume that this bug would probably also prevent restarting from ADIOS files. Is this correct? 
(I currently encounter restart issues with ADIOS on taurus ml - but I did not yet check the readability of the adios files.)

**Comment by KseniaBastrakova** [comment]
> Oh dang, I think the `7` should read `1` here:
> 
> https://github.com/ComputationalRadiationPhysics/picongpu/blob/5373b6e64decb304f42e25a4b69f06e51d38222a/include/picongpu/plugins/adios/writer/ParticleAttributeSize.hpp#L139
> 
> Copy-paste-upsi :)
> 
> @KseniaBastrakova do you like to change this and provide a PR for it?

Yes, I'm going to do that

**Comment by ax3l** [comment]
@KseniaBastrakova thanks, please feel free to ping me on the PR.

**Comment by sbastrakov** [comment]
@ax3l today @KseniaBastrakova is traveling, so I doubt.

**Comment by ax3l** [comment]
No problem, opened in #2312 because I need it myself, too.

Testing welcome, will go to sleep now ;)

**Comment by sbastrakov** [comment]
Fixed by #3120.

**Comment by ax3l** [comment]
Did you try this is readable with openPMD-api now?

**Comment by sbastrakov** [comment]
@ax3l good point, sorry for closing too early.

**Comment by ax3l** [comment]
Quick test on my data:
- [x] reading a series that only contains fields (before): ok
- [x] reading a series that only contains particles (before): fail with above error
- [x] reading a series that only contains fields (fixed): ok
- [ ] reading a series that only contains particles (before): waiting for data

**Comment by sbastrakov** [comment]
@ax3l have your test simulation finished?

**Comment by ax3l** [comment]
Still have this error:
```
ERROR: Attribute '/data/40000/particles/e_onLaserAxis/particles_info/unitSI' is not found!
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-4-08c242bf7112> in <module>
----> 1 s = api.Series("electrons_%T.bp", api.Access_Type.read_only)

RuntimeError: Internal error: Failed to get ADIOS1 attribute during attribute read
```

We could probably ignore this and throw a warning instead in openPMD-api... Although currently we do not encourage to store non-openPMD-data inside the reserved paths that are parsed, so this would not pass the validator nor reading of the standard anyway: https://github.com/openPMD/openPMD-standard/issues/115

The main problem here is that ADIOS particles do not yet use openPMD particle patches, I never had time to finish that last part in the ADIOS I/O set since it affects restarts and I already spend a lot of time making the backend safe under changing MPI distributions between runs.

**Comment by ax3l** [comment]
@sbastrakov I pushed work-around that ignores the `particles_info` object of PIConGPU to openPMD-api:
https://github.com/openPMD/openPMD-api/pull/635

Still your observation is valid: this object should not be at that location and violates openPMD, so we can leave this issue open until this is fixed.

**Comment by KseniaBastrakova** [comment]
thanks, @ax3l , now I ran into other problems [https://github.com/openPMD/openPMD-api/issues/636], after solving them I will try

**Comment by ax3l** [comment]
I just release `0.10.2` to aide you further and safe you from compiling yourself: https://github.com/openPMD/openPMD-api/issues/636#issuecomment-566757281

**Comment by KseniaBastrakova** [comment]
I still have this problem after update  0.10.2

>     series_hdf = openpmd_api.Series(adios_name, openpmd_api.Access_Type.read_only)
> RuntimeError: Unexpected Attribute datatype for 'timeOffset'
> kseniia@kseniia-Lenovo-V110-15ISK:~/Documents/particle_reduction/

**Comment by ax3l** [comment]
Confusing, I tested this on my own files and had no problems.

**Comment by KseniaBastrakova** [comment]
> Confusing, I tested this on my own files and had no problems.
> Did you write new files with PIConGPU and included the bugfix from PR #3120?

now, I'm pretty sure, that it was my mistake, sorry. 

I test it again, everything works and I had no problems

**Comment by ax3l** [comment]
Hooray, that's great! :)
